src/main/scala/faces/utils/Helpers.scala

```diff
@@ -287,15 +290,398 @@ case class Helpers(cfg: FacesSettings)(implicit rnd: Random) {
+    }
+    counter
+  }
+
+
+  def writeOcclusionToImages(image: PixelImage[RGBA], mask: PixelImage[RGBA], occlusion: PixelImage[RGBA], start: (Int, Int)): (PixelImage[RGBA], PixelImage[RGBA]) = {
+    val return_Image = PixelImage(image.width, image.height, (x, y) => {
+      if (x >= start._1 && x < start._1 + occlusion.width && y >= start._2 && y < start._2 + occlusion.height) {
+        val occlusion_rgba = occlusion(x - start._1, y - start._2)
+        if (occlusion_rgba.a > 0.5)
+          occlusion(x - start._1, y - start._2)
+        else
+          image(x, y)
+      }
+      else
+        image(x, y)
+    })
+    val return_Mask = PixelImage(mask.width, mask.height, (x, y) => {
+      if (x >= start._1 && x < start._1 + occlusion.width && y >= start._2 && y < start._2 + occlusion.height) {
+        val occlusion_rgba = occlusion(x - start._1, y - start._2)
+        if (occlusion_rgba.a > 0.5)
+          RGBA.apply(0, 0, 0, 0) // Change the color of the occlusion on the GROTRU mask
+        else
+          mask(x, y)
+      }
+      else
+        mask(x, y)
+    })
+    (return_Image, return_Mask)
+  }
+
+  def flipImage(image: PixelImage[RGBA], flip: (Boolean, Boolean, Boolean)): PixelImage[RGBA] = {
+    if (flip._3)
+      PixelImage(image.height, image.width, (x, y) => {
+        var x_new = x
+        var y_new = y
+        if (flip._1)
+          x_new = image.height - 1 - x
+        if (flip._2)
+          y_new = image.width - 1 - y
+        image(y_new, x_new)
+      })
+    else
+      PixelImage(image.width, image.height, (x, y) => {
+        var x_new = x
+        var y_new = y
+        if (flip._1)
+          x_new = image.width - 1 - x
+        if (flip._2)
+          y_new = image.height - 1 - y
+        image(x_new, y_new)
+      })
+  }
+
+  def getRandImage(onlyHands: Int): (Boolean, Boolean, Boolean, String, Int) = {
+    val r = scala.util.Random
+    val A = Array(
+      "hand_dark.png",
+      "hand_dark_2.png",
+      "hand_dark_3.png",
+      "hand_white.png",
+      "hand_white_1.png",
+      "hand_white_2.png",
+      "hand_white_3.png",
+      "hand_white_4.png",
+      "hand_white_5.png",
+      "hand_white_6.png",
+      "micro_1.png",
+      "micro_2.png",
+      "micro_3.png")
+    val index_hand_images = 0 to 9
+    val index_microphone_images = 10 to A.size - 1
+    if (onlyHands == 1) {
+      val image_index = r.nextInt(index_hand_images.size)
+      return (r.nextBoolean(), r.nextBoolean(), r.nextBoolean(), A(index_hand_images(image_index)),image_index)
+    }
+    else if (onlyHands == 2) {
+      val image_index = r.nextInt(index_microphone_images.size)
+      return (r.nextBoolean(), r.nextBoolean(), r.nextBoolean(), A(index_microphone_images(image_index)), image_index)
+    }
+    else {
+      val image_index = r.nextInt(A.size)
+      return (r.nextBoolean(), r.nextBoolean(), r.nextBoolean(), A(image_index), image_index)
+    }
+  }
+
+  def getEyeCenters(id: Int, n: Int): Array[Int] = {
```

Comment:
please use eye coordinates as rendered when landmarks are generated and don't read them from file again